src/inspector.rs

```diff
@@ -0,0 +1,675 @@
+                format!("Some({callback:p})")
+            } else {
+                "None".to_owned()
+            };
+
+        f.debug_struct("Inspector")
+            .field("raw", &self.raw)
+            .field("_on_response_callback", &on_response_str)
+            .field("_on_wait_frontend_message_on_pause_callback", &on_wait_str)
+            .finish()
+    }
+}
+
+/// The callback function used to send the messages back to the client
+/// of the inspector (to the one who is debugging).
+extern "C" fn on_response(
+    string: *const ::std::os::raw::c_char,
+    rust_callback: *mut ::std::os::raw::c_void,
+) {
+    let string = unsafe { std::ffi::CStr::from_ptr(string) }.to_string_lossy();
+    log::trace!("Outgoing message: {string}");
+    let rust_callback: &mut Box<OnResponseCallback> = unsafe {
+        &mut *(rust_callback as *mut std::boxed::Box<dyn std::ops::FnMut(std::string::String)>)
+    };
+    rust_callback(string.to_string())
+}
+
+extern "C" fn on_wait_frontend_message_on_pause(
+    raw: *mut crate::v8_c_raw::bindings::v8_inspector_c_wrapper,
+    rust_callback: *mut ::std::os::raw::c_void,
+) -> ::std::os::raw::c_int {
+    log::trace!("on_wait_frontend_message_on_pause");
+    let rust_callback: &mut Box<OnWaitFrontendMessageOnPauseCallback> = unsafe {
+        &mut *(rust_callback
+            as *mut std::boxed::Box<
+                dyn std::ops::FnMut(*mut crate::v8_c_raw::bindings::v8_inspector_c_wrapper) -> i32,
+            >)
+    };
+    rust_callback(raw)
+}
+
+impl Inspector {
+    /// Creates a new [Inspector].
+    pub fn new(
+        context: &V8ContextScope<'_, '_>,
+        on_response_callback: Box<OnResponseCallback>,
+        on_wait_frontend_message_on_pause_callback: Box<OnWaitFrontendMessageOnPauseCallback>,
+    ) -> Self {
+        let on_response_callback = Box::new(on_response_callback);
+        let on_response_callback = Box::into_raw(on_response_callback);
+        let on_wait_frontend_message_on_pause_callback =
+            Box::new(on_wait_frontend_message_on_pause_callback);
+        let on_wait_frontend_message_on_pause_callback =
+            Box::into_raw(on_wait_frontend_message_on_pause_callback);
+
+        let raw = unsafe {
+            crate::v8_c_raw::bindings::v8_InspectorCreate(
+                context.inner_ctx_ref,
+                Some(on_response),
+                on_response_callback as _,
+                Some(on_wait_frontend_message_on_pause),
+                on_wait_frontend_message_on_pause_callback as _,
+            )
+        };
+
+        let on_response_callback: Box<Box<dyn FnMut(String)>> =
+            unsafe { Box::from_raw(on_response_callback as *mut _) };
+        let on_wait_frontend_message_on_pause_callback =
+            unsafe { Box::from_raw(on_wait_frontend_message_on_pause_callback as *mut _) };
+        Self {
+            raw,
+            _on_response_callback: Some(on_response_callback),
```

Comment:
Why do we need to also save those pointer? I see that it is also not used (by the `_` prefix...)